File: web-admin/src/services/medicationService.js
import api from './api';

export const medicationService = {
  getAll: (page = 0, size = 10, search = '') =>
    api.get(`/medications?page=${page}&size=${size}&search=${search}`),
  getById: (id) => api.get(`/medications/${id}`),
  create: (data) => api.post('/medications', data),
  update: (id, data) => api.put(`/medications/${id}`, data),
  delete: (id) => api.delete(`/medications/${id}`),
};

File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/MedicationService.java
package vn.edu.fpt.doghandbook.backend.service;

import org.springframework.web.multipart.MultipartFile;
import vn.edu.fpt.doghandbook.backend.dto.request.MedicationRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.MedicationResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;

public interface MedicationService {

    PageResponse<MedicationResponse> getAll(int page, int size, String search, String status);

    MedicationResponse getById(Integer id);

    MedicationResponse create(MedicationRequest request, Integer createdByUserId, MultipartFile image);

    MedicationResponse update(Integer id, MedicationRequest request, Integer actorUserId, MultipartFile image);

    void delete(Integer id);
}
